File: ui/src/interfaces/incoming-document.ts
import { IncomingDocumentRow } from "./incoming-document-row";
import { Contragent } from "./contragent";
import { Contract } from "./contract";
import { Warehouse } from "./warehouse";

export interface IncomingDocument {
  id?: number;
  documentDate: string;
  documentNumber: string;
  posted: boolean;
  contragent?: Contragent;
  contract?: Contract;
  warehouse?: Warehouse;
  rows: IncomingDocumentRow[];
}

File: api/src/main/java/by/delaidelo/tests/testworks/domain/IncomingDocument.java
package by.delaidelo.tests.testworks.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
public class IncomingDocument {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate documentDate;
    private String documentNumber;
    private boolean posted;

    @ManyToOne
    private Contragent contragent;

    @ManyToOne
    private Contract contract;

    @ManyToOne
    private Warehouse warehouse;

    @OneToMany(mappedBy = "document", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<IncomingDocumentRow> rows;

}